fix: exclude django-helpdesk from openapi schema generation

## Observed Behavior
During OpenAPI schema generation (via `rtk bazel build //web:openapi_schema`), the `django-helpdesk` views trigger warnings and database errors because they execute queries on the unmigrated temporary database or lack explicit serializers for datatables/timeline views:

```
/home/phil/.cache/bazel/.../helpdesk/views/api.py:44: Error [TicketViewSet]: exception raised while getting serializer. Hint: Is get_serializer_class() returning None or is get_queryset() not working without a request? Ignoring the view for now. (Exception: no such table: helpdesk_customfield)
/home/phil/.cache/bazel/.../helpdesk/views/api.py:24: Error [UserTicketViewSet]: exception raised while getting serializer. Hint: Is get_serializer_class() returning None or is get_queryset() not working without a request? Ignoring the view for now. (Exception: no such table: helpdesk_customfield)
/home/phil/.cache/bazel/.../helpdesk/views/api.py:24: Warning [UserTicketViewSet]: could not derive type of path parameter "id" because it is untyped and obtaining queryset from the viewset failed. Consider adding a type to the path (e.g. <int:id>) or annotating the parameter type with @extend_schema. Defaulting to "string".
/home/phil/.cache/bazel/.../helpdesk/views/staff.py: Error [datatables_ticket_list]: unable to guess serializer. This is graceful fallback handling for APIViews. Consider using GenericAPIView as view base class, if view is under your control. Either way you may want to add a serializer_class (or method). Ignoring view for now.
/home/phil/.cache/bazel/.../helpdesk/views/staff.py: Error [timeline_ticket_list]: unable to guess serializer. This is graceful fallback handling for APIViews. Consider using GenericAPIView as view base class, if view is under your control. Either way you may want to add a serializer_class (or method). Ignoring view for now.
Warning: operationId "support_api_tickets_retrieve" has collisions [('/support/api/tickets/', 'get'), ('/support/api/tickets/{id}/', 'get')]. resolving with numeral suffixes.
Warning: operationId "support_api_user_tickets_retrieve" has collisions [('/support/api/user_tickets/', 'get'), ('/support/api/user_tickets/{id}/', 'get')]. resolving with numeral suffixes.
```

## Expected Behavior
The OpenAPI schema generation runs cleanly with zero warnings or errors originating from third-party application views like `django-helpdesk` (which are mounted under `/support/` and are not part of the PotterDoc API used by the frontend).

## Reproduction Steps
<!-- These steps were validated locally by the reporting agent -->
1. Clear the bazel build cache or force-rerun the OpenAPI schema generation target:
   ```bash
   rtk bazel build //web:openapi_schema
   ```
2. Observe the stderr output from the genrule execution.

**Result:**
The command outputs 39 errors (4 unique) and 3 warnings (3 unique), all originating from `helpdesk/views/api.py` and `helpdesk/views/staff.py`.

**Expected:**
The schema is generated cleanly with no errors or warnings.

## Environment
- Branch: `main`
- Relevant object state: None (occurs on clean/uncached bazel builds of `//web:openapi_schema`).

## Root Cause (preliminary)
In [urls.py](file:///home/phil/code/glaze/backend/urls.py#L110), the `django-helpdesk` URLs are included under `support/`:
```python
path("support/", include("helpdesk.urls")),
```
By default, `drf-spectacular` inspects all registered url patterns in the Django project. Since `django-helpdesk` endpoints are under `/support/`, they are parsed for documentation. However:
- Some helpdesk viewsets/serializers (like `TicketViewSet` and `UserTicketViewSet`) query `CustomField.objects.all()` inside their `__init__` methods, raising database exceptions (e.g. `no such table: helpdesk_customfield`) because schema generation runs against an unmigrated database.
- Some other views lack rest-framework attributes for guessing response schemas.

Since these endpoints are not part of the PotterDoc client/frontend API, they should be excluded from schema generation entirely.

## Proposed Solution
Register a preprocessing hook in `backend/settings.py` to filter endpoints so that only those starting with `/api/` are included:

1. Add a preprocessing filter function in [settings.py](file:///home/phil/code/glaze/backend/settings.py) (defining it directly in `settings.py` avoids needing to edit Bazel `BUILD` source files):
   ```python
   def openapi_preprocessing_filter_spec(endpoints):
       return [
           (path, path_regex, method, callback)
           for (path, path_regex, method, callback) in endpoints
           if path.startswith("/api/")
       ]
   ```
2. Register the hook in `SPECTACULAR_SETTINGS` in [settings.py](file:///home/phil/code/glaze/backend/settings.py):
   ```python
   SPECTACULAR_SETTINGS = {
       # ...
       "PREPROCESSING_HOOKS": ["backend.settings.openapi_preprocessing_filter_spec"],
   }
   ```

## Acceptance Criteria
- [ ] Schema generation via `rtk bazel build //web:openapi_schema` finishes without warnings or errors.
- [ ] The generated schema contains only endpoints matching the `/api/` prefix.

## Out of Scope
- Modifying `django-helpdesk` source files or running complete migrations on the temporary schema database.

--- a/api/tests/test_health.py
+++ b/api/tests/test_health.py
@@ -65,3 +65,12 @@ class TestHealthReady:
         request = rf.get("/api/health/ready/", HTTP_X_FORWARDED_HOST="159.223.154.68")
         response = health_ready(request)
         assert response.status_code == 200
+
+    def test_openapi_schema_contains_only_api_endpoints(self):
+        from drf_spectacular.generators import SchemaGenerator
+
+        generator = SchemaGenerator()
+        schema = generator.get_schema(request=None, public=True)
+        assert len(schema["paths"]) > 0
+        for path in schema["paths"].keys():
+            assert path.startswith("/api/")

--- a/backend/settings.py
+++ b/backend/settings.py
@@ -148,6 +148,15 @@ SIMPLE_JWT = {
     "AUTH_HEADER_TYPES": ("Bearer",),
 }
 
+
+def openapi_preprocessing_filter_spec(endpoints):
+    filtered = []
+    for path, path_regex, method, callback in endpoints:
+        if path.startswith("/api/"):
+            filtered.append((path, path_regex, method, callback))
+    return filtered
+
+
 SPECTACULAR_SETTINGS = {
     "TITLE": "PotterDoc API",
     "DESCRIPTION": (
@@ -181,7 +190,9 @@ SPECTACULAR_SETTINGS = {
             },
         }
     },
+    "PREPROCESSING_HOOKS": ["backend.settings.openapi_preprocessing_filter_spec"],
 }
+
 
 MIDDLEWARE = [
     "django.middleware.security.SecurityMiddleware",

--- a/backend/test_settings.py
+++ b/backend/test_settings.py
@@ -127,6 +127,15 @@ SIMPLE_JWT = {
     "AUTH_HEADER_TYPES": ("Bearer",),
 }
 
+
+def openapi_preprocessing_filter_spec(endpoints):
+    filtered = []
+    for path, path_regex, method, callback in endpoints:
+        if path.startswith("/api/"):
+            filtered.append((path, path_regex, method, callback))
+    return filtered
+
+
 SPECTACULAR_SETTINGS = {
     "TITLE": "PotterDoc API",
     "DESCRIPTION": (
@@ -160,7 +169,9 @@ SPECTACULAR_SETTINGS = {
             },
         }
     },
+    "PREPROCESSING_HOOKS": ["backend.test_settings.openapi_preprocessing_filter_spec"],
 }
+
 
 MIDDLEWARE = [
     "django.middleware.security.SecurityMiddleware",

